python_modules/dagster/dagster/core/storage/event_log/sql_event_log.py

```diff
@@ -652,152 +660,189 @@ def get_event_records(
+                row_by_asset_key[asset_key] = row
+                continue
+            materialization_or_event = (
+                deserialize_json_to_dagster_namedtuple(row[1]) if row[1] else None
+            )
+            if isinstance(materialization_or_event, EventLogEntry):
+                if asset_details.last_wipe_timestamp > materialization_or_event.timestamp:
+                    # this asset has not been materialized since being wiped, skip
+                    continue
+                else:
+                    # add the key
+                    row_by_asset_key[asset_key] = row
+            else:
+                row_by_asset_key[asset_key] = row
+                wiped_timestamps_by_asset_key[asset_key] = asset_details.last_wipe_timestamp
+
+        if wiped_timestamps_by_asset_key:
+            backcompat_query = (
+                db.select(
+                    [
+                        SqlEventLogStorageTable.c.asset_key,
+                        db.func.max(SqlEventLogStorageTable.c.timestamp),
+                    ]
+                )
+                .where(
+                    SqlEventLogStorageTable.c.asset_key.in_(
                         [
-                            SqlEventLogStorageTable.c.asset_key,
-                            db.func.max(SqlEventLogStorageTable.c.timestamp),
+                            asset_key.to_string()
+                            for asset_key in wiped_timestamps_by_asset_key.keys()
                         ]
                     )
-                    .where(
-                        SqlEventLogStorageTable.c.asset_key.in_(
-                            [asset_key.to_string() for asset_key in wiped_timestamps.keys()]
-                        )
-                    )
-                    .group_by(SqlEventLogStorageTable.c.asset_key)
-                    .order_by(db.func.max(SqlEventLogStorageTable.c.timestamp).asc())
-                ).fetchall()
+                )
+                .group_by(SqlEventLogStorageTable.c.asset_key)
+                .order_by(db.func.max(SqlEventLogStorageTable.c.timestamp).asc())
+            )
+            with self.index_connection() as conn:
+                backcompat_rows = conn.execute(backcompat_query).fetchall()
+            materialization_times = {
+                AssetKey.from_db_string(row[0]): row[1] for row in backcompat_rows
+            }
+            for asset_key, wiped_timestamp in wiped_timestamps_by_asset_key.items():
+                materialization_time = materialization_times.get(asset_key)
+                if not materialization_time or utc_datetime_from_naive(
+                    materialization_time
+                ) < utc_datetime_from_timestamp(wiped_timestamp):
+                    # remove rows that have not been materialized since being wiped
+                    row_by_asset_key.pop(asset_key)
+
+        has_more = limit and len(rows) == limit
```

Comment:
What if the limit happens to equal the number of rows in the table? Seems like `has_more` could be false when there's nothing more?

Reply:
Also, if I understand correctly, when the asset_keys, prefix, or cursor arguments are provided, there's a difference between the number of rows being fetched in the query and the number of rows in the table on the whole. Does that not cause a problem here? I.e. someone might fetch a particular prefix, we see all the assets fro that prefix are migrated, and then we mark the entire table as migrated, even though assets for other prefixes have not been migrated yet?

Reply:
yep, good catch... updated and added a test